features/netsocket/DNS.h

```diff
@@ -52,10 +69,11 @@ class DNS {
      *  The callback should not perform expensive operations such as socket recv/send
      *  calls or blocking operations.
      *
-     *  @param result  NSAPI_ERROR_OK on success, negative error code on failure.
+     *  @param result  Negative error code on failure or
+     *                 value that represents the number of DNS records
```

Comment:
Isn't this actually going to be a breaking change?
If someone was relying on the async callback to give him an `NSAPI_ERROR_OK`, they might have their code broken now, that the result contains the number of records...

Reply:
This is now breaking change, which I think is fine too, but needs to wait for Mbed OS 6, and requires migration section in the release notes.
@AnttiKauppila Please advice whether you want to refactor the API, or keep the backward compatibility.

I would recommend to refactor, but it is Antti's choice.

Reply:
I agree with this change. Let's put it then to go into Mbed OS 6